test/scripts/parallel-local/main.js

```diff
@@ -120,68 +141,87 @@ const handleWorkerExit = (code, projectName) => {
  * Displays the results of all tests by reading specific log files and aggregating results.
  * It first prints the contents of 'tests.log', calculates totals from 'results.log',
  * and finally reads 'fails.log' for any failures.
- * @returns {void} Does not return a value.
+ * @returns {void}
  */
-const displayResults = () => {
+const displayResults = async () => {
+    // eslint-disable-next-line no-console
     console.log('\nResults:\n');
-    testSuites.forEach((testSuiteName) => {
-        readLogFiles(testSuiteName, 'tests.log');
-    });
+    await readAllLogFiles('tests.log');
+
+    // eslint-disable-next-line no-console
     console.log('\n');
 
     let totalPassing = 0;
     let totalFailing = 0;
     let totalPending = 0;
 
-    testSuites.forEach((testSuiteName) => {
+    for (const testSuiteName of testSuites) {
         totalPassing += aggregateResults(testSuiteName, 'results.log', 'Passing');
         totalFailing += aggregateResults(testSuiteName, 'results.log', 'Failing');
         totalPending += aggregateResults(testSuiteName, 'results.log', 'Pending');
-    });
+    }
 
     // Calculate elapsed time
     const endTime = new Date();
     const elapsed = new Date(endTime - startTime);
     const minutes = elapsed.getUTCMinutes();
 
     // Display total passing with elapsed time in minutes and seconds
+    // eslint-disable-next-line no-console
     console.log('   ', totalPassing, 'Passing', `(${minutes}m)`);
+    // eslint-disable-next-line no-console
     console.log('   ', totalFailing, 'Failing');
     if (totalPending > 0) {
+        // eslint-disable-next-line no-console
         console.log('   ', totalPending, 'Pending');
     }
-
+    // eslint-disable-next-line no-console
     console.log('\n');
 
-    testSuites.forEach((testSuiteName) => {
-        readLogFiles(testSuiteName, 'fails.log');
-    });
-
-    cleanupTestDirectories(testSuites);
+    await readAllLogFiles('fails.log');
 };
 
 /**
- * Reads and logs the contents of a specified log file from the directory of each test suite.
+ * Asynchronously reads and logs the contents of a specified log file from the directory of each test suite.
+ * This function finds the log file in the test suite's directory, reads it if present, and logs the content.
+ * If the file is not found or an error occurs during reading, an error is logged.
+ *
  * @param {string} testSuiteName - The name of the test suite directory.
  * @param {string} logFileName - The name of the log file to read.
- * @returns {void} Does not return a value.
+ * @returns {Promise<void>} A Promise that resolves when the file has been read and logged, or an error has occurred.
  */
-const readLogFiles = (testSuiteName, logFileName) => {
-    const testDirectoryPath = path.join('./database/storage/', testSuiteName);
+const readLogFilesAsync = async (testSuiteName, logFileName) => {
```

Comment:
If the function is read, I expect it to return the content. I would rather name it `dumpLogFileContentToConsole` for example?

Reply:
Done